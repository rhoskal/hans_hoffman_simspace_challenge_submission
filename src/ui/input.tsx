import React from "react";
import styled from "styled-components";

type Props = {
  name: string;
  // eslint-disable-next-line no-undef
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
  placeholder?: string;
  type?: string;
  value?: string;
};

const StyledInput = styled.input`
  min-width: 300px;
  border-width: 1px;
  padding: 5px;
`;

export function Input({ name, onChange, placeholder = "Search", type = "text", value }: Props) {
  return (
    <>
      <label htmlFor={name} style={{ display: "none" }}>
        Breed
      </label>
      <StyledInput
        autoComplete="off"
        id={name}
        onChange={e => {
          onChange && onChange(e);
        }}
        placeholder={placeholder}
        type={type}
        value={value}
      />
    </>
  );
}
