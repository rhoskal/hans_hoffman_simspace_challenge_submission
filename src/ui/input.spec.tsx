import React from "react";
import { fireEvent, render } from "@testing-library/react";

import { Input } from "./input";

describe("[Component]: Input", function() {
  // eslint-disable-next-line no-undef
  let onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  beforeEach(function() {
    onChange = jest.fn();
  });

  it("should handle onChange", function() {
    const name = "test-input";
    const { getByPlaceholderText } = render(<Input onChange={onChange} name={name} value="" />);

    const inputNode = getByPlaceholderText("Search");
    fireEvent.change(inputNode, { target: { value: "asdf" } });

    expect(onChange).toHaveBeenCalled();
  });

  it("should match snapshot", function() {
    const { container } = render(<Input name="test-input" />);

    expect(container).toMatchSnapshot();
  });
});
