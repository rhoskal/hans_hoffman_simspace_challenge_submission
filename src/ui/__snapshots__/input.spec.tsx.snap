// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component]: Input should match snapshot 1`] = `
<div>
  <label
    for="test-input"
    style="display: none;"
  >
    Breed
  </label>
  <input
    autocomplete="off"
    class="sc-AxjAm fXbnfk"
    id="test-input"
    placeholder="Search"
    type="text"
    value=""
  />
</div>
`;
